macros/utils/cross_db_utils/hour_of_week.sql

```diff
@@ -0,0 +1,32 @@
+{% macro edr_hour_of_week_expression(date_expr) %}
+    {{ return(adapter.dispatch('edr_hour_of_week_expression','elementary')(elementary.edr_cast_as_date(date_expr))) }}
+{% endmacro %}
+
+{# Databricks, Spark: #}
+{% macro default__edr_hour_of_week_expression(date_expr) %}
+    concat(cast(DATE_FORMAT({{ date_expr }}, 'EEEE') as string), cast(HOUR({{ date_expr }}) as string))
```

Comment:
Can we use the [`edr_cast_as_string`](https://github.com/elementary-data/dbt-data-reliability/blob/2e95cf900ea263341d057c552fb6fb0a7dfe41b4/macros/utils/data_types/cast_column.sql#L17) macro? The current implementation does not seem to work on all data platforms.

Reply:
Heya! Thanks for the feedback because we're passing expressions with a jinja variable in it I think it's going to be cleaner to just use the logic that underpins `edr_cast_as_string()` if that's ok with you? 

Using the whole function I think is gonna end up being something like this unless I add another macro layer in. Could be a better way to do this but I'm not sure?
```
{% macro default__edr_hour_of_week_expression(date_expr) %}
    concat({{ elementary.edr_cast_as_string("DATE_FORMAT(" + date_expr + ", 'EEEE'))" }}, {{ elementary.edr_cast_as_string("HOUR(" + date_expr + ")" ) }})
{% endmacro %}
```

A cleaner approach seems to be just modifying my cast to use the `elementary.edr_type_string()` like this:
```{% macro default__edr_hour_of_week_expression(date_expr) %}
    concat(cast(DATE_FORMAT({{ date_expr }}, 'EEEE') as {{ elementary.edr_type_string() }}), cast(HOUR({{ date_expr }}) as {{ elementary.edr_type_string() }}))
{% endmacro %}``` Seems to keep things a bit cleaner as we're not passing a combination of string and variable components into another macro? 

I've put updated the PR to follow the second option of the above let me know your thoughts on this though!

Reply:
Sounds totally reasonable 👌